lib/utils/log/writer.go

```diff
@@ -79,7 +82,53 @@ func (s *FileSharedWriter) Reopen() (err error) {
 	return trace.Wrap(err)
 }
 
+// runWatcher spawns goroutine with the watcher loop to consume events of moving
+// or removing the log file to re-open it for log rotation purposes.
+func (s *FileSharedWriter) runWatcher() error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = watcher.Add(s.Name())
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I believe that the goroutine that services `watcher.Events` and `watcher.Errors` must be started before any method calls on `watcher`, or you risk a deadlock.